tracer/test/Datadog.Trace.IntegrationTests/SpanTagTests.cs

```diff
@@ -144,5 +117,143 @@ public void SpanSampler_ShouldTagMultiple_OnSpanFinish()
+            traces[0].Should().HaveCount(3); // ...with 3 spans
+
+            var rootSpan = traces[0].SingleOrDefault(s => s.ParentId is null or 0)!;
+            rootSpan.Should().NotBeNull();
+
+            // assert that root span has the span sampling tags
+            rootSpan.Tags.Should().Contain(Tags.SingleSpanSampling.RuleRate, expectedRuleRate);
+            rootSpan.Tags.Should().Contain(Tags.SingleSpanSampling.MaxPerSecond, expectedMaxPerSecond);
+            rootSpan.Tags.Should().Contain(Tags.SingleSpanSampling.SamplingMechanism, expectedSamplingMechanism);
+
+            // assert child spans have span sampling tags
+            var childSpans = traces[0].Where(s => s.ParentId is not null and not 0);
+
+            foreach (var span in childSpans)
+            {
+                span.Tags.Should().Contain(Tags.SingleSpanSampling.RuleRate, expectedRuleRate);
+                span.Tags.Should().Contain(Tags.SingleSpanSampling.MaxPerSecond, expectedMaxPerSecond);
+                span.Tags.Should().Contain(Tags.SingleSpanSampling.SamplingMechanism, expectedSamplingMechanism);
+            }
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldNotTag_WhenSpansAreKept()
+        {
+            var ruleRate = "1";
+            var maxPerSecond = "1000";
+            var samplingMechanism = "8";
+
+            using (var rootScope = _tracer.StartActive("root"))
+            {
+                using (var childScope = _tracer.StartActive("child1"))
+                {
+                }
+
+                using (var childScope = _tracer.StartActive("child2"))
+                {
+                }
+            }
+
+            var traces = _testApi.Wait();
+            traces.Should().HaveCount(1); // 1 trace...
+            traces[0].Should().HaveCount(3); // ...with 3 spans
+
+            var rootSpan = traces[0].SingleOrDefault(s => s.ParentId is null or 0)!;
+            rootSpan.Should().NotBeNull();
+
+            // assert that root span has the span sampling tags
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.RuleRate, ruleRate);
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.MaxPerSecond, maxPerSecond);
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.SamplingMechanism, samplingMechanism);
+
+            // assert child spans have span sampling tags
+            var childSpans = traces[0].Where(s => s.ParentId is not null and not 0);
+
+            foreach (var span in childSpans)
+            {
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.RuleRate, ruleRate);
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.MaxPerSecond, maxPerSecond);
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.SamplingMechanism, samplingMechanism);
+            }
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldNotTag_WhenTraceContext_IsNull()
```

Comment:
My first impression is that this is kind of an edge case that isn't representative of user code, so maybe we can delete this test? Even when we have spans who have a parent span from a different process (we use `SpanContext` to represent that), we'll start a new `TraceContext` as soon as call `StartSpan`, so I'm struggling to come up with a valid use case where there is no `TraceContext`. If there is one though, please correct me 😄

Reply:
Yeah I don't think this would be possible from a normal user perspective as the only way to actually make it happen is via `internal` calls from my understanding.
So deleting makes sense to me, thanks for pointing it out!

Reply:
I went through it again and yeah I don't see a way for it to be `null` unless it is manually setup that way, so removed in e7afde922f7a493e037a7926809031696f884630